android/src/main/java/io/sentry/react/RNSentryModule.java

```diff
@@ -343,15 +345,29 @@ public void captureScreenshot(Promise promise) {
             return;
         }
 
-        final byte[] raw = takeScreenshot(activity, logger, buildInfo);
-        if (raw == null) {
+        CountDownLatch doneSignal = new CountDownLatch(1);
+        final byte[][] bytesWrapper = {{}};
+        UiThreadUtil.runOnUiThread(() -> {
```

Comment:
M: Maybe we check first if its not in the UI Thread? Using `UiThreadUtil#isOnUiThread` so we don't do the overhead unnecessarily.

Reply:
Make sense.